featuretools/computational_backends/feature_tree.py

```diff
@@ -25,21 +27,62 @@ def __init__(self, entityset, features, ignored=None):
             self.ignored = set([])
         else:
             self.ignored = ignored
+
+        self.feature_hashes = set([f.hash() for f in features])
+
         all_features = {f.hash(): f for f in features}
-        feature_deps = {}
+        feature_dependencies = {}
+        feature_dependents = defaultdict(set)
         for f in features:
             deps = f.get_deep_dependencies(ignored=ignored)
-            feature_deps[f.hash()] = deps
+            feature_dependencies[f.hash()] = deps
             for dep in deps:
+                feature_dependents[dep.hash()].add(f.hash())
                 all_features[dep.hash()] = dep
-                feature_deps[dep.hash()] = dep.get_deep_dependencies(ignored=ignored)
+                subdeps = dep.get_deep_dependencies(
+                    ignored=ignored)
+                feature_dependencies[dep.hash()] = subdeps
+                for sd in subdeps:
+                    feature_dependents[sd.hash()].add(dep.hash())
+        # turn values which were hashes of features into the features themselves
+        # (note that they were hashes to prevent checking equality of feature objects,
+        #  which is not currently an allowed operation)
+        self.feature_dependents = {fhash: [all_features[dhash] for dhash in feature_dependents[fhash]]
```

Comment:
why do you iterate over the keys in `all_features` rather than `feature_dependents`?

Reply:
If a feature has no dependencies, it won't be put in feature_dependents by the above code. But, looking at it now, I can just explicitly add those in the for loop.

Reply:
hmm actually that doesn't make much sense because it's a defaultdict, so a feature with no deps will get treated as an empty value in the defaultdict anyway.